velox/functions/sparksql/String.h

```diff
@@ -455,4 +455,83 @@ struct SubstrFunction {
   }
 };
 
+template <typename T>
+struct OverlayFunction {
+  VELOX_DEFINE_FUNCTION_TYPES(T);
+
+  // ASCII input always produces ASCII result.
+  static constexpr bool is_default_ascii_behavior = true;
+
+  FOLLY_ALWAYS_INLINE void call(
+      out_type<Varchar>& result,
+      const arg_type<Varchar>& input,
+      const arg_type<Varchar>& replace,
+      const int32_t pos,
+      const int32_t len) {
+    doCall<false>(result, input, replace, pos, len);
+  }
+
+  FOLLY_ALWAYS_INLINE void callAscii(
+      out_type<Varchar>& result,
+      const arg_type<Varchar>& input,
+      const arg_type<Varchar>& replace,
+      const int32_t pos,
+      const int32_t len) {
+    doCall<true>(result, input, replace, pos, len);
+  }
+
+  template <bool isAscii>
+  FOLLY_ALWAYS_INLINE void doCall(
+      out_type<Varchar>& result,
+      const arg_type<Varchar>& input,
+      const arg_type<Varchar>& replace,
+      const int32_t pos,
+      const int32_t len) {
+    auto pair = substring<isAscii>(input, 1, pos - 1);
+    auto byteRange = stringCore::getByteRange<isAscii>(
+        input.data(), pair.first + 1, pair.second);
+    result.append(StringView(
+        input.data() + byteRange.first, byteRange.second - byteRange.first));
+    result.append(replace);
+
+    int32_t length = len >= 0 ? len : stringImpl::length<isAscii>(replace);
+    pair = substring<isAscii>(input, pos + length, INT32_MAX);
+    byteRange = stringCore::getByteRange<isAscii>(
+        input.data(), pair.first + 1, pair.second);
+    result.append(StringView(
+        input.data() + byteRange.first, byteRange.second - byteRange.first));
+  }
+
+  template <bool isAscii>
+  FOLLY_ALWAYS_INLINE std::pair<int32_t, int32_t> substring(
```

Comment:
Would you document this method to explain valid values for pos and length and how negative and large values are handled?

Reply:
Please, document the return value as well.